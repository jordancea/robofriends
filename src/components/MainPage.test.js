import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import MainPage from "./MainPage.js";

let wrapper;
beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: "",
    isPending: false
  };
  wrapper = shallow(<MainPage {...mockProps} />);
});

it("should render the MainPage component", () => {
  expect(toJson(wrapper)).toMatchSnapshot();
});

it("should filters robots corretcly", () => {
  expect(wrapper.instance().filterRobots()).toEqual([]);
  const mockProps2 = {
    onRequestRobots: jest.fn(),
    robots: [
      {
        id: 3,
        name: "John",
        email: "john@gmail.com"
      }
    ],
    searchField: "john",
    isPending: false
  };
  const wrapper2 = shallow(<MainPage {...mockProps2} />);
  expect(wrapper2.instance().filterRobots().length).toEqual(1);
});

it("should show loading", () => {
  const mockProps3 = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: "",
    isPending: true
  };
  const wrapper3 = shallow(<MainPage {...mockProps3} />);
  expect(wrapper3.find("h1.loading").length).toEqual(1);
});
