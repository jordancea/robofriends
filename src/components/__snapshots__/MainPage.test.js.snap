// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the MainPage component 1`] = `
<div
  className="tc"
>
  <h1
    className="f1"
  >
    RoboFriends
  </h1>
  <CounterButton
    color="red"
  />
  <SearchBox />
  <Scroll>
    <ErrorBoundary>
      <CardList
        robots={Array []}
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;
